variables.tf

```diff
@@ -3,31 +3,42 @@
 #
 # You must provide a value for each of these parameters.
 # ------------------------------------------------------------------------------
+variable "account_name_regex" {
+  type        = string
+  description = "The Terraform regular expression matching the name of the account(s) that you want to configure (e.g. \"^[[:alnum:]]-production$\").  See [https://www.terraform.io/language/functions/regex] for details on Terraform regular expression syntax."
+}
 
-variable "subnet_id" {
+variable "sso_admin_profile" {
   type        = string
-  description = "The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0)."
+  description = "The name of the AWS profile (typically found in your .aws/credentials file) to use for the default Terraform provider.  This profile's role must include permissions to administer Single Sign-On (SSO) resources.  For an example of a role like this, look at [https://github.com/cisagov/cool-accounts/pull/95]."
 }
 
 # ------------------------------------------------------------------------------
 # OPTIONAL PARAMETERS
 #
 # These parameters have reasonable defaults.
 # ------------------------------------------------------------------------------
-variable "ami_owner_account_id" {
-  type        = string
-  description = "The ID of the AWS account that owns the Example AMI, or \"self\" if the AMI is owned by the same account as the provisioner."
-  default     = "self"
-}
-
-variable "aws_availability_zone" {
-  type        = string
-  description = "The AWS availability zone to deploy into (e.g. a, b, c, etc.)."
-  default     = "a"
-}
 
 variable "aws_region" {
   type        = string
   description = "The AWS region to deploy into (e.g. us-east-1)."
   default     = "us-east-1"
 }
+
+variable "groups_to_add_access_to" {
+  type        = list(object({ group = string, permission_sets = list(string) }))
+  description = "A list of objects specifying Single Sign-On (SSO) groups to add permissions to.  Each object contains the SSO group name and the list of permission sets to add access to.  Example: [{ group = \"Admins\", permission_sets = [\"AWSAdministratorAccess\"] }]"
+  default     = []
+}
+
+variable "users_to_remove_access_from" {
+  type        = list(object({ username = string, permission_sets = list(string) }))
+  description = "A list of objects specifying Single Sign-On (SSO) users to remove permissions from.  Each object contains the SSO username and the list of permission sets to remove access from.  Example: [{ username = \"john.doe@example.com\", permission_sets = [\"AWSAdministratorAccess\"] }]"
+  default     = []
+}
```

Comment:
The code you have works for our purposes, but I could imagine a case where we might want to add access fro a particular user or remove access for a particular group.  Would it be difficult to have variables `groups_to_grant_access` and `users_to_grant_access` and
- Grant access for any groups or users that are listed
- Remove access for any groups or users that are not listed

I think this would
- Make the code more generally useful
- Give us two variables with similar intent, as opposed to two variables that are very similar but have opposite intent.  The latter is slightly confusing.

Reply:
In order to correctly grant/revoke access for each user and group, we need to be able to determine which users/groups currently have access to each permission set, but Terraform doesn't give us an easy way to fetch that info.  Terraform doesn't even have a way to grab all of the existing SSO user and group names.  It would have to be accomplished via AWS CLI calls and even then it will probably get messy (messier than this already is).

As a step towards what you're asking for, how would you feel about the addition of two more variables to cover all four cases?
* `groups_to_add_access_to`
* `groups_to_remove_access_from` (new)
* `users_to_add_access_to` (new)
* `users_to_remove_access_from`

Reply:
I would be fine with that, as long as it doesn't cause you a lot of extra work.

Reply:
Doing it this way is much less work than what you proposed- I don't think it will be very different from what I already have.

Reply:
I added the new features and re-requested your review.  Let me know what you think.

Reply:
Thanks for doing that.  Looks great!